GoogleSignIn/Tests/Unit/GIDSignInTest.m

```diff
@@ -503,6 +503,39 @@ - (void)testRestorePreviousSignInWhenSignedOut {
   [_authState verify];
 }
 
+- (void)testNotRestorePreviousSignInWhenSignedOutAndCompletionIsNil {
+  [[[_authorization expect] andReturn:_authState] authState];
+  [[[_authState expect] andReturnValue:[NSNumber numberWithBool:NO]] isAuthorized];
+
+  [_signIn restorePreviousSignInWithCompletion:nil];
+
+  XCTAssertNil(_signIn.currentUser);
+}
+
+- (void)testRestorePreviousSignInWhenCompletionIsNil {
+  [[[_authorization expect] andReturn:_authState] authState];
+  [[[_authState expect] andReturnValue:[NSNumber numberWithBool:YES]] isAuthorized];
+
+  OIDTokenResponse *tokenResponse =
+  [OIDTokenResponse testInstanceWithIDToken:[OIDTokenResponse fatIDToken]
+                                accessToken:kAccessToken
+                                  expiresIn:nil
+                               refreshToken:kRefreshToken
+                               tokenRequest:nil];
+
+  [[[_authState stub] andReturn:tokenResponse] lastTokenResponse];
+
+  [[[_user stub] andReturn:_user] alloc];
+  (void)[[[_user expect] andReturn:_user] initWithAuthState:OCMOCK_ANY
```

Comment:
Did you decide to not capture the values of `authState` and `profileData` here to compare against what is on `_signIn.currentUser` below? I think it would add value to the test.

Reply:
Looks like it's too tricky for this PR to disentangle the mocking. #306 tracks the issue. 👍